src/main/java/com/lambdaworks/redis/protocol/ConnectionWatchdog.java

```diff
@@ -194,7 +198,7 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
     /**
      * Schedule reconnect if channel is not available/not active.
      */
-    public synchronized void scheduleReconnect() {
+    public void scheduleReconnect() {
```

Comment:
After making state as `AtomicBoolean`, I think you mean we can remove this `synchronized`.